src/validation.h

```diff
@@ -744,6 +745,62 @@ bool InvalidateBlock(CValidationState& state, const CChainParams& chainparams, C
 /** Remove invalidity status from a block and its descendants. */
 void ResetBlockFailureFlags(CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+/**
+ * Serialized version of a UTXO set from which an assumed-valid CChainState
+ * can be constructed.
+ */
+class SnapshotMetadata
+{
+public:
+    //! The hash of the block that reflects the tip of the chain for the
+    //! UTXO set contained in this snapshot.
+    uint256 m_base_blockhash;
+
+    //! A hash of the snapshot contents as generated by `GetUTXOStats`.
+    uint256 m_utxo_contents_hash = {};
+
+    //! The number of coins in the UTXO set contained in this snapshot.
```

Comment:
In commit "validation: add unused SnapshotMetadata class" (3511d6b2ea65219a18dbb6d6ca45a4de62569dc1)

Comments for m_utxo_contents_hash m_coins_count fields could say what they are used for.

Reply:
Thanks for this feedback - made me realize that there's really no point anymore to bundling the contents hash with the snapshot.